tests/test_updater.py

```diff
@@ -886,7 +886,8 @@ def test_3__update_metadata_if_changed(self):
     self.repository_updater._update_metadata('timestamp', DEFAULT_TIMESTAMP_FILELENGTH)
     self.repository_updater._update_metadata_if_changed('snapshot', 'timestamp')
     self.repository_updater._update_metadata_if_changed('targets')
-    self.repository_updater._update_metadata_if_changed('root')
+    # TODO: should this still work?
+    # self.repository_updater._update_metadata_if_changed('root')
```

Comment:
I need to read through the spec and code to figure out whether this should still work, it throws:
```
Traceback (most recent call last):
  File "/home/joshuagl/Projects/tuf/tests/test_updater.py", line 890, in test_3__update_metadata_if_changed
    self.repository_updater._update_metadata_if_changed('root')
  File "/home/joshuagl/Projects/tuf/tuf/client/updater.py", line 1915, in _update_metadata_if_changed
    ['meta'][metadata_filename]
KeyError: 'root.json'
```

Since the removal of `root.json` from `snapshot.json`, which I believe makes sense but I want to double-check.

Reply:
Removal of this test is correct, the `root` metadata is no longer referenced by `snapshot`, hence the traceback above. Furthermore, the root role is updated separately and verified based on the existing root metadata, following the detailed client workflow logic in the spec, in the `_update_root_metadata()` function.